impl/src/main/java/org/jboss/cdi/tck/tests/deployment/discovery/EmptyBeansXmlDiscoveryTest.java

```diff
@@ -7,20 +7,21 @@
 import jakarta.enterprise.inject.spi.CDI;
 import org.jboss.arquillian.container.test.api.Deployment;
 import org.jboss.cdi.tck.AbstractTest;
-import org.jboss.shrinkwrap.api.ShrinkWrap;
-import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.cdi.tck.shrinkwrap.WebArchiveBuilder;
 import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.test.audit.annotations.SpecVersion;
 import org.testng.annotations.Test;
 
 /**
  * Tests that a singular archive with empty beans.xml results in annotated discovery mode.
  */
+@SpecVersion(spec = "cdi", version = "4.0")
 public class EmptyBeansXmlDiscoveryTest extends AbstractTest {
 
     @Deployment
     public static WebArchive createTestArchive() {
-        return ShrinkWrap.create(WebArchive.class).addClasses(SomeAnnotatedBean.class, SomeUnannotatedBean.class)
-                .addAsWebInfResource(new StringAsset(""), "beans.xml");
+        return new WebArchiveBuilder().withTestClassPackage(EmptyBeansXmlDiscoveryTest.class)
+                .withBeansXml("beans.xml").build();
```

Comment:
Shouldn't this be `withBeansXml("")`?

Reply:
Maybe both work? If you look at the usages of this method, all other tests use the syntax I have here

Reply:
Ah I get it now. What you have is correct.

It's just that the `withBeansXml` javadoc is super-confusing and its source code, without looking further, doesn't help either. The `String` parameter is a path to the classpath resource. It would seem you don't have to specify that, because the javadoc says:

```
Add <code>beans.xml</code> located in src/main/resource/{testPackagePath}.
```

but you do. The `beans.xml` from the javadoc refers to where the file is _placed into_, not where it is _read from_.

It should read something like this IMHO:

```
Add a file with given {@code beansXml} name located in src/main/resource/{testPackagePath} to the archive as <code>beans.xml</code>.
```

Sorry for the noise.